### PR TITLE
Added isChannelMember middleware

### PR DESCRIPTION
**Pull request by criticic**
Issue: #46

### CODE REVIEW

File: src/middlewares/isChannelMember.middleware.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const Channel = require('../models/channel.models.js');
+const {response_403, response_404, response_500} = require('../utils/responseCodes.utils.js');
+
+export async function isChannelMember(req, res, next) {
+    try {
+        const userId = req.body.user;
```
Comment (automatefreely) [inline src/middlewares/isChannelMember.middleware.js]: req.body.user this gives the whole mongo object

**Review by automatefreely — COMMENTED** [review]
You need to also allow admin of workspace

File: src/middlewares/isChannelMember.middleware.js (hunk 1 above)
Comment (criticic) [reply src/middlewares/isChannelMember.middleware.js]: oh okay done

File: src/middlewares/isChannelMember.middleware.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    try {
+        const userId = req.body.user._id;
+        const channelId = parseInt(req.params.id);
+        const channel = await Channel.findById(channelId).populate('workspace');
+        if(!channel) {
+            return response_404(res, 'Channel not found');
+        }
+
+        const user = channel.members.find(member => member._id === userId);
+        if(!user || channel.workspace.createdBy !== user._id) {
```
Comment (automatefreely) [inline src/middlewares/isChannelMember.middleware.js]: Here is an issue here.   -> "user._id"

Comment (criticic) [reply src/middlewares/isChannelMember.middleware.js]: fixed

**Pull request merged**